interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -301,36 +389,50 @@ function($log, $cacheFactory, $q, Sample, Activity, NodeInfo, MathFuncts) {
         return null;
       }
 
-      // (1) we obtain a list of nodes by retrieving node activity
-      //     for the first sample in our volcano plot
-      var firstSampleNodes = this.activityCache.get(sg['group-a'][0]);
-      // (2a) next, we build a new array (`retval`) comprised of `nodeObject`s
-      //      by walking through the `firstSampleNodes` and constructing a
-      //      `nodeObject` for each. [outer .map()]
-      var retval = firstSampleNodes.map(function(val, i, arr) {
-        var mapSampleIdsToActivity = function(sampId, i, arr) {
-          // (2b) the array of activity for each node is built by plucking the
-          //      activity `.value` for each sample within this node from the
-          //      `activityCache` [inner .map()]
-          // FIXME: counting on array order to match node order here
-          return this.activityCache.get(sampId)[val.node - 1].value;
-        };
-        var nodeObject = {
-          'id': val.node, // TODO: map this id to node name (via vega spec?)
-          'activityA': sg['group-a'].map(mapSampleIdsToActivity, this),
-          'activityB': sg['group-b'].map(mapSampleIdsToActivity, this)
-        };
-        nodeObject.diff = (
-          MathFuncts.mean(nodeObject.activityB) -
-          MathFuncts.mean(nodeObject.activityA)
-        );
-        nodeObject.logsig = -Math.log10(MathFuncts.tTest(
-          nodeObject.activityA, nodeObject.activityB
-        ).pValue());
+      // (1a) we obtain a list of nodes by retrieving node activity
+      //      for the first sample in our volcano plot
+      var firstSampleNodes = this.activityCache.get(sg['group-a'][0]).map(
+        function(val) {
+          return val.node;  // extract just the node IDs
+        }
+      );
+      // (1b) now obtain (and cache) a name for each node id
+      var nodeInfoSetPromise = this.getNodeInfoSetPromise(firstSampleNodes);
```

Comment:
Let me see if I can find what I read that finally clarified for me how `this` works in ES5, because that clarity is what I'm reinforcing by doing it this way. You're right that I can use `cbSampleBin` in place of `this`, but it is unnecessary because the code in this line does not run in the context of a callback.

- [ ] look up reference for `this` in ES5